tests/test_src/test_provider/test_apiresponse.cpp

```diff
@@ -0,0 +1,100 @@
+#include "gtest/gtest.h"
+
+#include "provider/apiresponse.h"
+
+TEST(ErdGenericApiResponse, constructor_validData)
+{
+    EXPECT_NO_THROW(ErdGenericApiResponse apiResponse(R"({"data":"something","error":"some error","code":"some code"})"));
+}
+
+TEST(ErdGenericApiResponse, constructor_invalidData)
+{
+    EXPECT_THROW({
+                     try
+                     {
+                         ErdGenericApiResponse apiResponse(R"(Invalid json)");
+                     }
+                     catch(const std::invalid_argument &e)
+                     {
+                         EXPECT_EQ( ERROR_MSG_JSON_SERIALIZED + R"(Invalid json)", e.what() );
+                         throw;
+                     }
+                 }, std::invalid_argument );
+}
+
+TEST(ErdGenericApiResponse, getters_allExist)
+{
+    ErdGenericApiResponse apiResponse(R"({"data":{"account":{"address":"erd1qq"}},"error":"some error","code":"some code"})");
+
+    nlohmann::json expectedData;
+    expectedData["account"]["address"] = "erd1qq";
+
+    EXPECT_EQ(apiResponse.getCode(), "some code");
+    EXPECT_EQ(apiResponse.getError(), "some error");
+    EXPECT_EQ(apiResponse.getData<nlohmann::json>(), expectedData);
```

Comment:
For a more beautiful test, can you define a struct instead of using `nlohmann::json`,  and then call `getData<MyStruct>()`?

Or just rewrite the assertion to not use `nlohmann::json` at all - here and in the tests below? (optionally).

Reply:
As discussed yesterday, we will keep it as it is